action/protocol/registry.go

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) 2019 IoTeX
+// This is an alpha (internal) release and is not suitable for production. This source code is provided 'as is' and no
+// warranties are given as to title or non-infringement, merchantability or fitness for purpose and, to the extent
+// permitted by law, all liability for your use of the code is disclaimed. This source code is governed by Apache
+// License 2.0 that can be found in the LICENSE file.
+
+package protocol
+
+import (
+	"sync"
+
+	"github.com/pkg/errors"
+
+	"github.com/iotexproject/iotex-core/pkg/log"
+)
+
+// Registry is the hub of all protocols deployed on the chain
+type Registry struct {
+	protocols sync.Map
```

Comment:
should we use ptr to avoid pass sync map by copy?

Reply:
Instead, passing Registry ptr around?

Reply:
that works too. then Registry should only allow to pass by ptr. maybe we should add a comment for it